android/app/src/main/cpp/audio_engine.cpp

```
@@ -117,6 +120,112 @@ static void emitAudioErrorEvent(oboe::Result error) {
+// the ring at a 20 ms cadence and runs the JNI dispatch on its own
+// (long-attached) JNIEnv. The worker is owned by the engine — it starts in
+// `start()` after the streams open and stops in `stop()` before the streams
+// close, so an in-flight audio callback cannot push to a dead queue.
+static TalkingEventQueue g_talkingQueue;
+static std::atomic<bool> g_talkingWorkerStop{false};
+static std::thread g_talkingWorkerThread;
+
+// JNI dispatch for a single VAD edge. Runs on the worker thread, NOT the
+// audio thread — taking g_jniMutex and attaching to the JVM here is fine.
+static void emitTalkingEventJni(bool talking) {
+    JavaVM* jvm = nullptr;
```

Clang (14.0.6)

[warning] variable 'jvm' is not initialized

(cppcoreguidelines-init-variables)